applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -367,13 +380,32 @@ impl AppStateInner {
         Ok(())
     }
 
+    pub async fn refresh_connected_peers_state(&mut self) -> Result<(), UiError> {
+        let connections = self.wallet.comms.connectivity().get_active_connections().await?;
+
+        let mut peers = Vec::new();
+        for c in connections.iter() {
+            if let Ok(p) = self.wallet.comms.peer_manager().find_by_node_id(c.peer_node_id()).await {
+                peers.push(p);
+            }
+        }
+
+        self.data.connected_peers = peers;
```

Comment:
nit: Single allocation, clone peer manager once

```suggestion
        let peer_manager = self.wallet.comms.peer_manager();
        let mut peers = Vec::with_capacity(connections.len());
        for c in connections {
            let peer = peer_manager.find_by_node_id(c.peer_node_id()).await?;
            peers.push(peer);
        }

        self.data.connected_peers = peers;
```